### PR TITLE
added jobs-bulletin watchdog

### LINKED ISSUE #344: Add #jobs-bulletin listener.
As suggested in the #uqcs-meta channel, set up a listener for posts to #jobs-bulletin that posts a follow up message there directing all questions/related discussion to #jobs-discussion. 

Message should be a generic notice that the channel is for job postings only so as to not be out of place should the poster decide upon seeing it to remove their message. 

Additionally, a message should be sent to poster explaining the purpose of the bulletin (and outlining any rules/suggestions for posts there - i.e. fair work compliance, and a warning about what to expect on posting an 'unpaid role').

### PR DESCRIPTION
**Pull request by nicklambourne**
As per title. Fixes #344. Feedback welcome.

- Messages user with #jobs-bulletin rules when they join.
- Reminds users who post in #jobs-bulletin of the rules.
- Messages #jobs-bulletin directing conversation to DM or #jobs-discussion.

### CODE REVIEW

**Comment by nicklambourne** [comment]
On posting in channel example:
screen shot 2018-09-06 at 11 28 57 am

**Comment by nicklambourne** [comment]
Also wondering if it would be possible to make a subprocess that sticks around for ~min after someone posts to delete the chat redirection message if the original message is deleted?

**Comment by mitchmcdee** [comment]
message wording looks good 👍 could "Only employers advertising jobs" be reworded though maybe? "Only job advertisements"?

will give a code review "soon"

File: uqcsbot/scripts/jobs.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    user = bot.users.get(evt.get("user"))
+
+    if user is None or user.is_bot:
+        return
+
+    bot.post_message(jobs_bulletin, f"{user.display_name} has posted a new job in <#{jobs_bulletin.id}>! "
+                                    f":tada: \nPlease ask any questions in <#{jobs_discussion.id}> or in a private "
+                                    f"message to <@{user.user_id}|{user.display_name}>")
+
+    # Send new user their welcome messages.
```
Comment (mitchmcdee) [inline uqcsbot/scripts/jobs.py]: fix this comment-- this function isn't about new users :^)

**Comment by mitchmcdee** [comment]
re: "Also wondering if it would be possible to make a subprocess that sticks around for ~min after someone posts to delete the chat redirection message if the original message is deleted?"

Sounds definitely possible but 10x more work + effort, save it for another PR :^) if you think its super valuable chuck up an issue, but personally I don't think its worth

**Comment by jdcaperon** [comment]
I'll second the change to  `Only employers advertising jobs`.

File: uqcsbot/scripts/jobs.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    "*Rules for Everyone* \n1. Only employers advertising jobs are allowed to post in this channel.\n"
+    "2. All discussion about the posted jobs must take place in the #jobs-discussion channel or by direct message "
+    "with the person posting the advertisement. Please be respectful when interacting with companies and sponsors.",
+    "*Additional Rules for Employers Posting Jobs/Internship Opportunities:*\n"
+    f"3. We take the rights of our members and associates seriously. If you are posting an unpaid position, please be"
+    f" up front about the lack of remuneration and *mindful of* <{FAIR_WORK_INTERNSHIPS_INFO}|*your obligations*> "
+    "under the _Fair Work Act (2009)_ :fairwork:. \n_tldr: if an intern (whether called that or not) adds value to "
+    "(or 'does productive work' for) your business, they must be remunerated with either a fair wage or equity_. If you"
+    " ignore these warnings, please expect to face criticism from the community (we will protect our members from being"
+    " exploited).",
```
Comment (jdcaperon) [inline uqcsbot/scripts/jobs.py]: Is it worth mentioning that unpaid positions should be approved by UQ if the applicant is a student within the EAIT school? https://www.eait.uq.edu.au/engineering-professional-practice-unpaid-placements

**Pull request merged**